### PR TITLE
For SyncVar, add nonblocking get, nonblocking take, put with timeout, an...

### PR DESCRIPTION
**Pull request by shawjef3**
I find myself having to use "get(1)" to get the current value (or lack thereof) of a SyncVar with minimal waiting. It'd be simpler if I could skip the 1 millisecond wait.

### CODE REVIEW

**Comment by adriaanm** [comment]
review by @phaller or @axel22

**Comment by adriaanm** [comment]
Note the syntax error:

```
src/library/scala/concurrent/SyncVar.scala:143: error: not found: value putNow
   val putNow(x : A): Boolean = synchronized {
        ^
```

**Comment by adriaanm** [comment]
Welcome, @shawjef3! Please [sign the Scala CLA](http://typesafe.com/contribute/cla/scala) and fix the typo. We'll also need some tests to go with this change, preferably of the JUnit kind (example: https://github.com/scala/scala/blob/master/test/junit/scala/collection/PriorityQueueTest.scala)

File: src/library/scala/concurrent/SyncVar.scala [hunk 1]
```diff
@@ -89,6 +108,45 @@ class SyncVar[A] {
...
+    }
+  }
+
+  /** Sets this SyncVar's value without waiting. If it already is defined,
+  *   does nothing.
+  *
+  *  @param x           the value to store in this SyncVar
+  *  @return            `true` if this SyncVar was set, `false` otherwise.
+  */
+  val putNow(x : A): Boolean = synchronized {
```
Comment (axel22) [inline src/library/scala/concurrent/SyncVar.scala:143]: Is this supposed to be a `def`?

Comment (axel22) [inline src/library/scala/concurrent/SyncVar.scala:143]: As Adriaan said, please take time to read the contributor instructions and add tests.

**Comment by axel22** [comment]
generally looks ok to me, just needs tests and needs to compile.

**Comment by retronym** [comment]
Closing for now, please resubmit if you get a chance to address the review comments.

**Pull request closed without merging**